### PR TITLE
🔒️ Internal audit/lp

### PR DESCRIPTION
**Pull request by j-vp**

### CODE REVIEW

File: test/fuzz/LendingPool/DoActionWithLeverage.fuzz.t.sol [hunk 1]
```diff
@@ -245,10 +247,8 @@ contract DoActionWithLeverage_LendingPool_Fuzz_Test is LendingPool_Fuzz_Test {
         vm.assume(liquidity >= amountLoaned);
         vm.assume(amountLoaned > 0);
 
-        uint256 fee = uint256(amountLoaned) * originationFee / 10_000;
-
         vm.prank(users.creatorAddress);
-        pool.setOriginationFee(originationFee);
```
Comment (Thomas-Smets) [inline test/fuzz/LendingPool/DoActionWithLeverage.fuzz.t.sol:251]: Why fee to 0?

File: test/fuzz/LendingPool/SetMaxInitiatorFee.fuzz.t.sol [hunk 2]
```diff
@@ -37,7 +37,7 @@ contract SetMaxLiquidationFees_LendingPool_Fuzz_Test is LendingPool_Fuzz_Test {
         emit MaxLiquidationFeesSet(maxFeeInitiation, maxFeeClosing);
         pool.setMaxLiquidationFees(maxFeeInitiation, maxFeeClosing);
 
-        assertEq(pool.getMaxInitiatorFee(), maxFeeInitiation);
-        assertEq(pool.getMaxClosingFee(), maxFeeClosing);
+        assertEq(pool.getmaxInitiationFee(), maxFeeInitiation);
```
Comment (Thomas-Smets) [inline test/fuzz/LendingPool/SetMaxInitiatorFee.fuzz.t.sol:40]:
```suggestion
        assertEq(pool.getMaxInitiationFee(), maxFeeInitiation);
```

File: test/fuzz/LendingPool/SetMaxInitiatorFee.fuzz.t.sol [hunk 3]
```diff
@@ -37,7 +37,7 @@ contract SetMaxLiquidationFees_LendingPool_Fuzz_Test is LendingPool_Fuzz_Test {
...
+        assertEq(pool.getmaxTerminationFee(), maxFeeClosing);
```
Comment (Thomas-Smets) [inline test/fuzz/LendingPool/SetMaxInitiatorFee.fuzz.t.sol:41]:
```suggestion
        assertEq(pool.getMaxTerminationFee(), maxFeeClosing);
```

File: test/utils/Extensions.sol [hunk 4]
```diff
@@ -160,12 +160,12 @@ contract LendingPoolExtension is LendingPool {
         return fixedLiquidationCost;
     }
 
-    function getMaxInitiatorFee() public view returns (uint80) {
-        return maxInitiatorFee;
+    function getmaxInitiationFee() public view returns (uint80) {
```
Comment (Thomas-Smets) [inline test/utils/Extensions.sol:163]:
```suggestion
    function getMaxInitiationFee() public view returns (uint80) {
```

File: test/utils/Extensions.sol [hunk 5]
```diff
@@ -196,12 +196,12 @@ contract LendingPoolExtension is LendingPool {
         return tranches[id];
     }
 
-    function getAccountFactory() public view returns (address) {
-        return accountFactory;
+    function getACCOUNT_FACTORY() public view returns (address) {
```
Comment (Thomas-Smets) [inline test/utils/Extensions.sol:199]: function names without all caps

File: src/LendingPool.sol [hunk 6]
```diff
@@ -37,9 +37,9 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
     // Seconds per year, leap years ignored.
     uint256 internal constant YEARLY_SECONDS = 31_536_000;
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:38]: Missing Constants header

File: src/LendingPool.sol [hunk 7]
```diff
@@ -187,24 +189,26 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
     /**
      * @notice The constructor for a lending pool.
      * @param riskManager_ The address of the new Risk Manager.
-     * @param asset_ The underlying ERC-20 token of the Lending Pool.
+     * @param asset_ The underlying ERC20 token of the Lending Pool.
      * @param treasury_ The address of the protocol treasury.
-     * @param accountFactory_ The address of the Account Factory.
-     * @param liquidator_ The address of the Liquidator.
+     * @param ACCOUNT_FACTORY_ The address of the Account Factory.
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:194]: input is a variable -> small letters

File: src/LendingPool.sol [hunk 8]
```diff
@@ -187,24 +189,26 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
+     * @param LIQUIDATOR_ The address of the Liquidator.
      * @dev The name and symbol of the DebtToken are automatically generated, based on the name and symbol of the underlying token.
      */
-    constructor(address riskManager_, ERC20 asset_, address treasury_, address accountFactory_, address liquidator_)
+    constructor(address riskManager_, ERC20 asset_, address treasury_, address ACCOUNT_FACTORY_, address LIQUIDATOR_)
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:198]: Don't use capitals for input parameters

File: src/LendingPool.sol [hunk 9]
```diff
@@ -706,16 +680,17 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
      * @dev In case you accidentally sent funds to the pool, contact the current treasury manager.
      */
     function skim() external processInterests {
-        //During auctions, debt tokens are burned at start of the auction, while auctions proceeds are only returned
-        //at the end of the auction -> skim function must be blocked during auctions.
+        //During auction initiation, debt tokens representing the liquidation incentives are minted at start of the auction
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:683]:
```suggestion
        // During auction initiation, debt tokens representing the liquidation incentives are minted at start of the auction
```

File: src/LendingPool.sol [hunk 10]
```diff
@@ -827,37 +801,35 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
     /**
-     * @notice Sets the maxInitiatorFee.
-     * @param maxInitiatorFee_ The maximum fee that is paid to the initiator of a liquidation.
-     * @param maxClosingFee_ The maximum fee that is paid to the closer of a liquidation.
-     * @dev The liquidator sets the % of the debt that is paid to the initiator of a liquidation.
-     * This fee is capped by the maxInitiatorFee.
+     * @notice Sets the maxInitiationFee and maxTerminationFee.
+     * @param maxInitiationFee_ The maximum fee that is paid to the initiator of a liquidation.
+     * @param maxTerminationFee_ The maximum fee that is paid to the closer of a liquidation.
+     * @dev The liquidator sets the % of the debt that is paid to the initiator and terminator of a liquidation.
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:807]: Now we use both closer and terminator

File: src/LendingPool.sol [hunk 11]
```diff
@@ -870,30 +842,27 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
         // Increase the realised liquidity for the initiator.
-        realisedLiquidityOf[initiator] += liquidationInitiatorReward;
-        totalRealisedLiquidity = uint128(totalRealisedLiquidity + liquidationInitiatorReward);
         // The other incentives will only be added as realised liquidity for the respective actors
-        // After the auction is finished.
+        // after the auction is finished.
+        realisedLiquidityOf[initiator] += liquidationInitiatorReward;
+        totalRealisedLiquidity = SafeCastLib.safeCastTo128(totalRealisedLiquidity + liquidationInitiatorReward);
+
+        // If only ongoing auction, inform most Jr tranche that auctions are ongoing
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:863]: rephrase

File: src/LendingPool.sol [hunk 12]
```diff
@@ -938,22 +907,22 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
-        if (surplus > 0) realisedLiquidityOf[IAccount(account).owner()] += surplus;
+        // Pay out the "liquidationPenalty" to the LPs and Treasury.
+        _syncLiquidationFeeToLiquidityProviders(liquidationPenalty);
 
         // Unsafe cast: sum will revert if it overflows.
-        totalRealisedLiquidity = uint128(totalRealisedLiquidity + auctionTerminationReward + liquidationFee + surplus);
+        totalRealisedLiquidity =
+            uint128(totalRealisedLiquidity + auctionTerminationReward + liquidationPenalty + surplus);
 
         // Decrement the number of auctions in progress.
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:919]: move comment down

File: test/fuzz/LendingPool/DoActionWithLeverage.fuzz.t.sol (hunk 1 above)
Comment (j-vp) [reply test/fuzz/LendingPool/DoActionWithLeverage.fuzz.t.sol:251]: To simplify this test which just checks for an event emittal, fees are tested in other tests.

File: src/LendingPool.sol [hunk 13]
```diff
@@ -187,24 +193,26 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
...
+     * @param asset_ The underlying ERC20 token of the Lending Pool.
      * @param treasury_ The address of the protocol treasury.
-     * @param accountFactory_ The address of the Account Factory.
-     * @param liquidator_ The address of the Liquidator.
+     * @param account_factory The address of the Account Factory.
+     * @param LIQUIDATOR_ The address of the Liquidator.
      * @dev The name and symbol of the DebtToken are automatically generated, based on the name and symbol of the underlying token.
      */
-    constructor(address riskManager_, ERC20 asset_, address treasury_, address accountFactory_, address liquidator_)
+    constructor(address riskManager_, ERC20 asset_, address treasury_, address account_factory, address LIQUIDATOR_)
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:202]:
```suggestion
    constructor(address riskManager_, ERC20 asset_, address treasury_, address accountFactory, address liquidator)
```

File: src/LendingPool.sol [hunk 14]
```diff
@@ -399,19 +376,17 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
     /**
      * @notice Donate assets to the Lending Pool.
      * @param trancheIndex The index of the tranche to donate to.
-     * @param assets The amount of assets of the underlying ERC-20 tokens being deposited.
+     * @param assets The amount of assets of the underlying ERC20 tokens being deposited.
      * @dev Can be used by anyone to donate assets to the Lending Pool.
      * It is supposed to serve as a way to compensate the jrTranche after an
-     * auction didn't get sold and was manually Liquidated by the Protocol.
+     * auction didn't get sold and was manually Liquidated by the Protocol after cutoffTime.
```
Comment (Thomas-Smets) [inline src/LendingPool.sol:382]:
```suggestion
     * auction didn't get sold and was manually Liquidated after cutoffTime.
```

**Review by Thomas-Smets — APPROVED** [review]
gtg

**Pull request merged**